backend/src/main/java/com/votogether/domain/post/service/PostService.java

```diff
@@ -365,5 +369,50 @@ private <T, R> List<R> transformElements(final List<T> elements, final Function<
                 .toList();
     }
 
+    @Transactional(readOnly = true)
+    public List<PostRankingResponse> getRanking() {
+        final List<Post> posts = postRepository.findAllByClosingTypeAndSortTypeAndCategoryId(
+                PostClosingType.ALL,
+                PostSortType.HOT,
+                null,
+                PageRequest.of(0, BASIC_PAGING_SIZE)
+        );
+
+        final Map<Post, Integer> rankings = calculateRanking(posts);
+
+        return rankings.entrySet().stream()
+                .sorted(Comparator.comparingInt(entry -> entry.getValue()))
+                .map(entry ->
+                        new PostRankingResponse(
+                                PostCompactResponse.of(entry.getKey()),
+                                entry.getValue()
+                        )
+                )
+                .toList();
+    }
+
+    private Map<Post, Integer> calculateRanking(final List<Post> posts) {
+        final List<Post> sortedPosts = posts.stream()
+                .sorted(Comparator.comparingLong(post -> -post.getTotalVoteCount()))
+                .toList();
+
+        final Map<Post, Integer> rankings = new HashMap<>();
+
+        int currentRanking = 1;
+        int previousRanking = -1;
+        long previousVoteCount = -1;
+        for (Post post : sortedPosts) {
+            final long currentVoteCount = post.getTotalVoteCount();
+            final int ranking = (currentVoteCount == previousVoteCount) ? previousRanking : currentRanking;
+
+            rankings.put(post, ranking);
+
+            previousRanking = ranking;
+            previousVoteCount = currentVoteCount;
+            currentRanking++;
+        }
+
```

Comment:
`P2`
```suggestion
final Map<Post, Integer> rankings = new HashMap<>();
int ranking = 1;
long previousVoteCount = Long.MAX_VALUE;

for (Post post : sortedPosts) {
    final long currentVoteCount = post.getTotalVoteCount();
    if (currentVoteCount != previousVoteCount) {
        ranking++;
    }
    rankings.put(post, ranking);
    
    previousVoteCount = currentVoteCount;
}

```
이렇게하면 로직을 조금 단순화 할 수 있을 것 같아요

Reply:
`P3`
위의 로직에서 첫 게시글은 랭킹이 `2`가 되는거 아닌가요? 🤔

가독성은 너무 좋은 것 같습니다 👍

Reply:
좋은 제안 감사합니다!
if문안에 ++이 있어서 첫 게시글 랭킹을 0으로 수정해보았는데
중복순위가 있을때  1 1 1 4 5 가 아닌 1 1 1 2 3 이 되더라고요 ㅠ

Reply:
그러네요 일단 이 로직은 추후에 리팩터링 해보는 걸로 해야할 것 같네요..ㅎㅎ